bam_ampliconclip.c

```diff
@@ -938,6 +996,32 @@ static int bam_clip(samFile *in, samFile *out, samFile *reject, char *bedfile,
 
     if (file_open) {
         fclose(stats_fp);
+        file_open = 0;
+    }
+
+    if (param->primer_counts_file) {
+        if ((bed_count_summary_fp = fopen(param->primer_counts_file, "w")) == NULL) {
+            fprintf(stderr, "[ampliconclip] warning: cannot write count summary to %s.\n", param->primer_counts_file);
+        } else {
+            file_open = 1;
+        }
+
+        //print out the number of reads for each bed entry, bedgraph format
+        fprintf(bed_count_summary_fp, "#CHR\tLEFT\tRIGHT\tNAME\tSCORE\tSTRAND\tNUM_CLIPPED\n");
+        khiter_t itr;
+        for (itr = kh_begin(bed_hash); itr != kh_end(bed_hash); ++itr) {
+            if (kh_exist(bed_hash, itr)) {
+                sites = &kh_val(bed_hash, itr);
+                for (int i = 0; i < sites->length; i++) {
+                    fprintf(bed_count_summary_fp, "%s\t%"PRId64"\t%"PRId64"\t%s\t%s\t%s\t%"PRId64"\n",
+                        kh_key(bed_hash, itr), sites->bp[i].left, sites->bp[i].right, sites->bp[i].name,
+                        sites->bp[i].score, (sites->bp[i].rev ? "-" : "+") , sites->bp[i].num_reads);
```

Comment:
bp[i].rev can be uninitialised if the **--strand** option is not used.

Reply:
I've added this, but i think it's a bit strange that we modify the bed information... though this does accurately describe the count (which is not stranded now)
```
 char* strand_out = param->use_strand ? (sites->bp[i].rev ? "-" : "+") : ".";
```
I could change the struct to store a char instead of rev (more invasive, but maybe nicer than an int ?)

Reply:
The rev in the struct is used by ampliconstats as well as ampliconclip so I would prefer not to change it.

You could add another member to the structure to store the strand symbol (if present).

Reply:
Do you think it's better to 
1) store the char in the struct (which could in theory be out of sync with rev) 
or 
2) recalc like this and switch the strand column to '.' indicating that the count is unstranded ?
(i implemented #2, but not sure if that's best).

Reply:
Hmm, leave it the second way for now.  I can always change it later if need be.